training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1073,635 @@ Expected Output:
+$ ./tutorial -nl 8
+The chunk of A owned by Locale 0 is: 1 2
+The chunk of A owned by Locale 6 is: 13 14
+The chunk of A owned by Locale 2 is: 5 6
+The chunk of A owned by Locale 7 is: 15 16
+The chunk of A owned by Locale 3 is: 7 8
+The chunk of A owned by Locale 5 is: 11 12
+The chunk of A owned by Locale 4 is: 9 10
+The chunk of A owned by Locale 1 is: 3 4
 ```
-We see the number of `put`s has decreased drastically, and it's way faster!
 
-If we decrease `SIZE`, we see the benefit of the aggregation is outweighed by the cost of setting it up.
-You can play around with different values to see at what point aggregation becomes worthwhile for this example.
+<a id="aggregation"></a>
+## Aggregation
+
+For the best performance we want to minimize how often we move data between locales. But there are situations
+where you need to move lots of data between locales. For example assigning local values into a distributed array
+at random indices. In this case, one locale might be copying multiple elements to every other locale.
+Sending each of those elements one at a time results in a lot of communication. Aggregation is used to mitigate
+this.
+
+I like to think of this in terms of transporting people, where our goal is minimizing the total number of trips.
+If a bunch of people are going to the same place, they could all travel there individually. But we'd have
+fewer total trips if we wait until we have enough people to fill a bus and send a whole group at once.
+
+[Copy aggregation](https://chapel-lang.org/docs/modules/packages/CopyAggregation.html#copyaggregation)
+is the same idea. Instead of copying each value individually, we wait until we have a bunch
+that are all going to the same locale and copy them over all at once.
+
+There are two types of copy aggregation:
+* copying local values into remote variables (`DstAggregator`)
+* copying remote values into local variables (`SrcAggregator`)
+
+The question of which aggregator do I need to use can sometimes trip people up.
+This is because the same operation could use either a source or destination aggregator
+depending on which locale is performing the operation.
+
+For example:
+> Let's say `x` is stored on `locale_i` and we want to copy `x` into a position on `locale_j`.
+Which aggregator do we use? It depends
+> 
+> If `locale_i` is the one doing the computation, then `x` is a local value.
+Since we're viewing this from perspective of `locale_i`, positions on `locale_j` are remote.
+So we need a `DstAggregator` because we are putting a local value into a remote position.
+>
+> If instead `locale_j` is the one doing this computation, the position
+we want to write into is local.
+And from `locale_j`'s perspective `x` is remote. 
+Since we're getting a remote value and writing it into a local position, we need a `SrcAggregator`.
+
+It's important to note copy aggregation will only work if at least one side is local.
+Both sides being remote (remote-to-remote aggregation) is not supported.
```

Comment:
You could say "not currently supported" here, if desired (in the sense that I believe it could be added if someone really wanted/needed it in the future).